File: T1000&IT8951 SDK-3.0.0-20260120/xt-t1000-mcu-demo-3.0.0/app/src/main/java/com/xingtai/t1000/demo/ui/base/BaseActivity.java
package com.xingtai.t1000.demo.ui.base;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.view.LayoutInflater;


import com.xingtai.t1000.demo.util.TUtils;

import java.lang.reflect.Method;

import androidx.appcompat.app.AppCompatActivity;
import androidx.viewbinding.ViewBinding;

/**
 * TODO
 *
 * @author Kelly
 * @version 1.0.0
 * @filename BaseActivity
 * @time 2021/9/11 14:40
 * @copyright(C) 2021 song
 */
public abstract class BaseActivity<VB extends ViewBinding> extends AppCompatActivity {
    protected VB viewBinding;
    protected Context mContext;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        try {
            Method inflate = TUtils.getClass(getClass()).getDeclaredMethod("inflate", LayoutInflater.class);
            viewBinding = (VB) inflate.invoke(null, getLayoutInflater());
            setContentView(viewBinding.getRoot());
        } catch (Exception e) {
            e.printStackTrace();
        }
        mContext = this;
        ready();
        initVM();
        initView();
        initListener();
        initData();

    }

    protected void ready() {
    }

    protected void initVM() {
    }


    protected abstract void initView();

    protected abstract void initListener();

    protected abstract void initData();



    public void openActivity(Class clz) {
        openActivity(clz, null);
    }

    public void openActivity(Class clz, Bundle bundle) {
        Intent intent = new Intent(mContext, clz);
        if (bundle != null) {
            intent.putExtras(bundle);
        }
        mContext.startActivity(intent);
    }



    /**
     * 判断Activity是否Destroy
     *
     * @param mActivity
     * @return
     */
    public static boolean isDestroy(Activity mActivity) {
        if (mActivity == null || mActivity.isFinishing() || (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1 && mActivity.isDestroyed())) {
            return true;
        } else {
            return false;
        }
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        viewBinding = null;
    }

}

File: T1000&IT8951 SDK-3.0.0-20260120/xt-epd-deivce-demo/app/src/main/java/com/xingtai/epd/device/demo/ui/base/BaseActivity.kt
package com.xingtai.epd.device.demo.ui.base

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.view.LayoutInflater
import androidx.appcompat.app.AppCompatActivity
import androidx.viewbinding.ViewBinding

/**
 *
 *
 * @author Kelly
 * @version 1.0.0
 * @filename BaseActivity
 * @time 2021/9/11 14:40
 * @copyright(C) 2021 song
 */
abstract class BaseActivity<VB : ViewBinding>(private val inflate: (LayoutInflater) -> VB) : AppCompatActivity() {
    protected lateinit var mViewBinding: VB
    protected lateinit var mContext: Context
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mContext = this
        mViewBinding = inflate(layoutInflater)
        setContentView(mViewBinding.root)
        ready()
        initVM()
        initView()
        initListener()
        initData()
    }


    protected fun ready() {}
    protected fun initVM() {}
    protected abstract fun initView()
    protected abstract fun initListener()
    protected abstract fun initData()

    @JvmOverloads
    fun openActivity(clz: Class<*>, bundle: Bundle? = null) {
        val intent = Intent(mContext, clz)
        if (bundle != null) {
            intent.putExtras(bundle)
        }
        mContext.startActivity(intent)
    }



    companion object {
        /**
         * Determine whether the activity has been destroyed
         *
         * @param mActivity
         * @return
         */
        fun isDestroy(mActivity: Activity?): Boolean {
            return if (mActivity == null || mActivity.isFinishing || Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1 && mActivity.isDestroyed) {
                true
            } else {
                false
            }
        }
    }
}
